### PR TITLE
Migrate most of docs-client to hooks

### PR DESCRIPTION
**Pull request by anuraaga**
This doesn't migrate the largest file, `DebugPage.tsx` since it's already large enough without that :) Will migrate that in another PR.

Also removes `immutability-helper`, it was only used in one place which is easy enough with normal javascript.

### CODE REVIEW

**Comment by trustin** [comment]
I'm not a front expert. @Ta2Rim, are you interested in reviewing this pull request?

**Comment by ikhoon** [comment]
FYI(especially server side developer) to review this PR 😉(cc @minwoox @trustin )

## React Core
* [React Hooks: UseEffect, UseCallback, UseMemo - DEV Community 👩‍💻👨‍💻](https://dev.to/devcord/react-hooks-useeffect-usecallback-usememo-3o42)
* [React Top-Level API – React.memo](https://reactjs.org/docs/react-api.html#reactmemo)
* [Hooks API Reference – React.useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer)
*  [How to use useReducer in React Hooks for performance optimization](https://medium.com/crowdbotics/how-to-use-usereducer-in-react-hooks-for-performance-optimization-ecafca9e7bf5) 

## React Echosystem
* [https://material-ui.com/styles/api/#makestyles-styles-options-hook](https://material-ui.com/styles/api/#makestyles-styles-options-hook) 

## Typescript
* [The unknown Type in TypeScript — Marius Schulz](https://mariusschulz.com/blog/the-unknown-type-in-typescript)
The unknown type is only assignable to the any type and the unknown type itself.
* [Object Access: Bracket-Notation vs. Dot-Notation With TypeScript In Angular 2 RC 4](https://www.bennadel.com/blog/3126-object-access-bracket-notation-vs-dot-notation-with-typescript-in-angular-2-rc-4.htm)
```ts
interface OpenServices {
 [name: string]: boolean;
}
```

File: docs-client/src/containers/EnumPage/index.tsx [hunk 1]
```diff
@@ -37,58 +37,57 @@ interface OwnProps {
...
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {data.values.length > 0 ? (
+              data.values.map((value) => (
+                <TableRow key={value.name}>
+                  <TableCell>
+                    <code>{value.name}</code>
+                  </TableCell>
+                  {hasIntValue && <TableCell>Int Value</TableCell>}
```
Comment (ikhoon) [inline docs-client/src/containers/EnumPage/index.tsx]: I wonder why you revert this code?

**Comment by Ta2Rim** [comment]
> I'm not a front-end expert. @Ta2Rim, are you interested in reviewing this pull request?

Sure, I'd love to!
Let me check.

File: docs-client/src/containers/EnumPage/index.tsx (hunk 1 above)
Comment (anuraaga) [reply docs-client/src/containers/EnumPage/index.tsx]: Oops thanks for finding this!

**Pull request merged**